### PR TITLE
Fix shell script lint check errors

### PR DESCRIPTION
**Pull request by tellison**
This fixes a number of warnings and errors from running the shell lint checker.
There are some real bugs (e.g. use of undefined vars), but most are safety features; notably 
1. using quotes around var expansion to avoid word splitting and globbing by the shell,
1. fail safe when `cd`'ing to a directory, especially where we `cd` then `rf -f`

Special bonus addition is the image in our README.md to show the testing status.

### CODE REVIEW

File: git-hg/git-hg.sh [hunk 1]
```diff
@@ -16,41 +16,41 @@
...
-cd openjdk_git
+cd openjdk_git || exit
 git init
 # convert mercurial to github
-$WORKSPACE/fast-export/hg-fast-export.sh -r $WORKSPACE/openjdk_hg
+"$WORKSPACE"/fast-export/hg-fast-export.sh -r "$WORKSPACE"/openjdk_hg
 for i in corba jaxp jaxws langtools jdk hotspot nashorn; do
 	# copy the additional repos in
-	cp -r $WORKSPACE/openjdk_hg/$i $i
+	cp -r "$WORKSPACE"/openjdk_hg/$i $i
```
Comment (neomatrix369) [inline git-hg/git-hg.sh]: Why not use "${...} something ${i}" format? reads better and also safe

File: git-hg/git-hg.sh [hunk 2]
```diff
@@ -16,41 +16,41 @@
...
 done
 git checkout
 # remove remaining mercurial stuff
 rm -rf .hg .hgignore .hgtags README-builds.html README get_source.sh
 # create new README
-cp $WORKSPACE/openjdk-build/git-hg/README.md .
+cp "$WORKSPACE"/openjdk-build/git-hg/README.md .
```
Comment (neomatrix369) [inline git-hg/git-hg.sh]: same here as above - this will be in line with the guideline on https://github.com/AdoptOpenJDK/openjdk-build/pull/54

File: git-hg/git-hg.sh [hunk 3]
```diff
@@ -16,41 +16,41 @@
...
 echo ".hg/" > .gitignore
 git add .
 git commit -m "merge sources"
 git remote add releases git@github.com:AdoptOpenJDK/openjdk-jdk8u.git
 git fetch --all
 # check if git diff is the same
-if [ `git diff releases/master | wc -l` -gt 0 ]; then
+if [ "$(git diff releases/master | wc -l)" -gt 0 ]; then
```
Comment (neomatrix369) [inline git-hg/git-hg.sh]: :+1:

File: makejdk.sh [hunk 4]
```diff
@@ -27,15 +27,16 @@
 REPOSITORY=AdoptOpenJDK/openjdk-jdk8u
 OPENJDK_REPO_NAME=openjdk
 
-OS_KERNEL_NAME=$(echo $(uname) | awk '{print tolower($0)}')
+OS_KERNEL_NAME=$(uname | awk '{print tolower($0)}')
+OS_CPU_NAME=$(uname -m)
```
Comment (neomatrix369) [inline makejdk.sh]: minor: `uname` terminology would be machine or machine name

File: makejdk.sh [hunk 5]
```diff
@@ -141,67 +142,65 @@ setDefaultIfBranchIsNotProvided()
...
 }
 
 cloneOpenJDKGitRepo()
 {
-  echo $git
+  echo "${git}"
   if [ -d "${WORKING_DIR}/${OPENJDK_REPO_NAME}/.git" ] && [ "$REPOSITORY" == "AdoptOpenJDK/openjdk-jdk8u" ] ; then
     # It does exist and it's a repo other than the AdoptOpenJDK one
-    cd $WORKING_DIR/$OPENJDK_REPO_NAME
+    cd "${WORKING_DIR}/${OPENJDK_REPO_NAME}" || return
```
Comment (neomatrix369) [inline makejdk.sh]: :+1: I was referring to this convention above

File: makejdk.sh [hunk 6]
```diff
@@ -141,67 +142,65 @@ setDefaultIfBranchIsNotProvided()
...
     echo "${info}Will reset the repository at $PWD in 10 seconds...${git}"
     sleep 10
     echo "${git}Pulling latest changes from git repo"
     git fetch --all
     git reset --hard origin/$BRANCH
-    echo $normal
-    cd $WORKING_DIR
+    echo "${normal}"
+    cd "${WORKING_DIR}" || return
```
Comment (neomatrix369) [inline makejdk.sh]: We had a discussion on one of the PR that "${}" is the same as ${}, i noted it for our guideline (see https://github.com/AdoptOpenJDK/openjdk-build/pull/54)

File: git-hg/git-hg.sh [hunk 7]
```diff
@@ -16,41 +16,41 @@
 git clone https://github.com/frej/fast-export.git
```
Comment (neomatrix369) [inline git-hg/git-hg.sh]: Any reason for not setting the `set -eu` at the first line, this will make the `|| exit` and `|| return` unnecessary if we want to script to fail-fast, i would vote for fail-fast. Unless you want a graceful exit for a good reason, there could be disadvantages to graceful exits in some cases.

File: sbin/signalhandler.sh [hunk 8]
```diff
@@ -1,6 +1,27 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# For terminal colors
+esc=$(echo -en "\033")
```
Comment (neomatrix369) [inline sbin/signalhandler.sh]: We use these values repeatedly across multiple scripts, idea: extract into a single .sh file and source that file from all the scripts, will help maintain consistency and a central place to change or add new colour codes

File: sbin/signalhandler.sh [hunk 9]
```diff
@@ -1,6 +1,27 @@
...
+error="${esc}[0;31m"
+normal=$(echo -en "${esc}[m\017")
+
+
 exit_script() {
     if [[ -z ${KEEP} ]] ; then
-      docker ps -a | awk '{ print $1,$2 }' | grep $CONTAINER | awk '{print $1 }' | xargs -I {} docker rm -f {}
+      docker ps -a | awk '{ print $1,$2 }' | grep "$CONTAINER" | awk '{print $1 }' | xargs -I {} docker rm -f {}
```
Comment (neomatrix369) [inline sbin/signalhandler.sh]: maybe use {} instead of "", all depends on which convention we settle down for

File: makejdk.sh [hunk 10]
```diff
@@ -27,15 +27,16 @@
...
-BUILD_TYPE=${BUILD_TYPE:-normal}
-
-BUILD_FULL_NAME=$OS_KERNEL_NAME-$OS_MACHINE-$BUILD_TYPE-$JVM_VARIANT-release
+JVM_VARIANT=server
+if [[ "$OS_CPU_NAME" == "s390x" ]] || [[ "$OS_CPU_NAME" == "armv7l" ]] ; then
+ JVM_VARIANT=zero
+fi 
 
+BUILD_TYPE=normal
+BUILD_FULL_NAME=$OS_KERNEL_NAME-$OS_CPU_NAME-$BUILD_TYPE-$JVM_VARIANT-release
```
Comment (neomatrix369) [inline makejdk.sh]: I used {} for each of the variables in my PR

File: makejdk.sh [hunk 11]
```diff
@@ -141,67 +142,65 @@ setDefaultIfBranchIsNotProvided()
 setWorkingDirectoryIfProvided()
 {
   if [ -z "${WORKING_DIR}" ] ; then
-    echo "${info}WORKING_DIR is undefined so setting to $PWD${normal}"
+    echo "${info}WORKING_DIR is undefined so setting to $PWD${normal}."
```
Comment (neomatrix369) [inline makejdk.sh]: `{}` around `PWD` ==> `${PWD}`

File: makejdk.sh [hunk 12]
```diff
@@ -141,67 +142,65 @@ setDefaultIfBranchIsNotProvided()
...
     WORKING_DIR=$PWD
   else
-    echo "${info}Working dir is $WORKING_DIR${normal}"
+    echo "${info}Working dir is $WORKING_DIR${normal}."
```
Comment (neomatrix369) [inline makejdk.sh]: same as above for WORKING_DIR

File: makejdk.sh [hunk 13]
```diff
@@ -141,67 +142,65 @@ setDefaultIfBranchIsNotProvided()
...
+    docker run --privileged -t -v "${WORKING_DIR}/${OPENJDK_REPO_NAME}":/openjdk/jdk8u/openjdk --entrypoint jtreg.sh "${CONTAINER}"
   fi
 }
 
 buildAndTestOpenJDKViaDocker()
 {
-  PS_DOCKER=$(ps -ef | grep "docker" | wc -l)
-
-  if [ -z $(which docker) ] || [ ${PS_DOCKER} -lt 2 ]; then
+  if [ -z "$(which docker)" ] || [ -z "$(pgrep "docker")" ]; then
```
Comment (neomatrix369) [inline makejdk.sh]: :+1:

File: makejdk.sh [hunk 14]
```diff
@@ -219,33 +218,33 @@ buildAndTestOpenJDKViaDocker()
...
-     echo $good
+     echo "$good"
      docker ps -a | awk '{ print $1,$2 }' | grep $CONTAINER | awk '{print $1 }' | xargs -I {} docker rm -f {}
      docker build -t $CONTAINER docker/jdk8u/x86_64/ubuntu
-     echo $normal
+     echo "$normal"
   fi
 
-  docker run --privileged -t -v $WORKING_DIR/$OPENJDK_REPO_NAME:/openjdk/jdk8u/openjdk --entrypoint build.sh $CONTAINER
+  docker run --privileged -t -v "$WORKING_DIR/$OPENJDK_REPO_NAME:/openjdk/jdk8u/openjdk" --entrypoint build.sh "$CONTAINER"
```
Comment (neomatrix369) [inline makejdk.sh]: `{}` for all the ENV variables in this line?

File: makejdk.sh [hunk 15]
```diff
@@ -219,33 +218,33 @@ buildAndTestOpenJDKViaDocker()
...
 
   testOpenJDKViaDocker
 
   CONTAINER_ID=$(docker ps -a | awk '{ print $1,$2 }' | grep openjdk_container | awk '{print $1 }'| head -1)
 
   if [[ "${COPY_TO_HOST}" == "true" ]] ; then
-    echo "Copying to the host with docker cp $id:/openjdk/jdk8u/OpenJDK.tar.gz $TARGET_DIR"
-    docker cp $CONTAINER_ID:/openjdk/jdk8u/OpenJDK.tar.gz $TARGET_DIR
+    echo "Copying to the host with docker cp $CONTAINER_ID:/openjdk/jdk8u/OpenJDK.tar.gz $TARGET_DIR"
```
Comment (neomatrix369) [inline makejdk.sh]: {} for the ENV variables, instead of the ""

File: makejdk.sh [hunk 16]
```diff
@@ -219,33 +218,33 @@ buildAndTestOpenJDKViaDocker()
...
+    docker cp "$CONTAINER_ID":/openjdk/jdk8u/OpenJDK.tar.gz "$TARGET_DIR"
   fi
 
   if [[ "${JTREG}" == "true" ]] ; then
     echo "Copying jtreg reports from docker"
-    docker cp $CONTAINER_ID:/openjdk/jdk8u/jtreport.zip $TARGET_DIR
-    docker cp $CONTAINER_ID:/openjdk/jdk8u/jtwork.zip $TARGET_DIR
+    docker cp "$CONTAINER_ID":/openjdk/jdk8u/jtreport.zip "$TARGET_DIR"
+    docker cp "$CONTAINER_ID":/openjdk/jdk8u/jtwork.zip "$TARGET_DIR"
```
Comment (neomatrix369) [inline makejdk.sh]: same for this and all the above changed lines

File: makejdk.sh [hunk 17]
```diff
@@ -258,14 +257,14 @@ testOpenJDKInNativeEnvironmentIfExpected()
 {
   if [[ "$JTREG" == "true" ]];
   then
-      $WORKING_DIR/sbin/jtreg.sh $WORKING_DIR $OPENJDK_REPO_NAME $BUILD_FULL_NAME $JTREG_TEST_SUBSETS
+      "$WORKING_DIR"/sbin/jtreg.sh "$WORKING_DIR" "$OPENJDK_REPO_NAME" "$BUILD_FULL_NAME" "$JTREG_TEST_SUBSETS"
```
Comment (neomatrix369) [inline makejdk.sh]: {} instead of ""

File: makejdk.sh [hunk 18]
```diff
@@ -283,7 +282,7 @@ buildAndTestOpenJDK()
 
 initialiseEscapeCodes
 sourceSignalHandler
-parseCommandLineArgs $@
+parseCommandLineArgs "$@"
```
Comment (neomatrix369) [inline makejdk.sh]: :+1:

File: makejdk.sh [hunk 19]
```diff
@@ -258,14 +257,14 @@ testOpenJDKInNativeEnvironmentIfExpected()
...
-      $WORKING_DIR/sbin/jtreg.sh $WORKING_DIR $OPENJDK_REPO_NAME $BUILD_FULL_NAME $JTREG_TEST_SUBSETS
+      "$WORKING_DIR"/sbin/jtreg.sh "$WORKING_DIR" "$OPENJDK_REPO_NAME" "$BUILD_FULL_NAME" "$JTREG_TEST_SUBSETS"
   fi
 }
 
 buildAndTestOpenJDKInNativeEnvironment()
 {
   echo "Calling sbin/build.sh $WORKING_DIR $TARGET_DIR $OPENJDK_REPO_NAME $BUILD_FULL_NAME $JVM_VARIANT"
-  $WORKING_DIR/sbin/build.sh $WORKING_DIR $TARGET_DIR $OPENJDK_REPO_NAME $BUILD_FULL_NAME $JVM_VARIANT
+  "$WORKING_DIR"/sbin/build.sh "$WORKING_DIR" "$TARGET_DIR" "$OPENJDK_REPO_NAME" "$BUILD_FULL_NAME" "$JVM_VARIANT"
```
Comment (neomatrix369) [inline makejdk.sh]: same here

File: sbin/build.sh [hunk 20]
```diff
@@ -89,9 +93,9 @@ checkingAndDownloadingAlsa()
   if [[ ! -z "${FOUND_ALSA}" ]] ; then
     echo "Skipping ALSA download"
   else
-    wget -nc ftp://ftp.alsa-project.org/pub/lib/alsa-lib-$ALSA_LIB_VERSION.tar.bz2
-    tar xvf alsa-lib-$ALSA_LIB_VERSION.tar.bz2
-    rm alsa-lib-$ALSA_LIB_VERSION.tar.bz2
+    wget -nc ftp://ftp.alsa-project.org/pub/lib/alsa-lib-"$ALSA_LIB_VERSION".tar.bz2
+    tar xvf alsa-lib-"$ALSA_LIB_VERSION".tar.bz2
+    rm alsa-lib-"$ALSA_LIB_VERSION".tar.bz2
```
Comment (neomatrix369) [inline sbin/build.sh]: {} instead of {} for all three lines

File: sbin/build.sh [hunk 21]
```diff
@@ -105,19 +109,20 @@ checkingAndDownloadingFreetype()
...
 
-    if [ $(uname -m) = "ppc64le" ]; then
+    if [ "$OS_CPU_NAME" = "ppc64le" ]; then
+      # shellcheck disable=SC1083
       PARAMS="--build=$(rpm --eval %{_host})"
     fi
 
     # We get the files we need at $WORKING_DIR/installedfreetype
-    bash ./configure --prefix=$WORKING_DIR/$OPENJDK_REPO_NAME/installedfreetype $PARAMS && make all && make install
+    bash ./configure --prefix="$WORKING_DIR"/"$OPENJDK_REPO_NAME"/installedfreetype "$PARAMS" && make all && make install
```
Comment (neomatrix369) [inline sbin/build.sh]: {} in place of "" for all changed lines above

File: sbin/jtreg_prep.sh [hunk 22]
```diff
@@ -116,10 +116,10 @@ fi
 # Step 3: Unpack Java if we need to.
 
 if [[ "$JAVA_SOURCE" == *\.tar\.gz ]]; then #If it's a tar file, unpack it.
-  cd "${JAVA_DESTINATION}"
-  tar xf *.tar.gz
+  cd "${JAVA_DESTINATION}" || exit
+  tar xf ./*.tar.gz
   echo "Java has been untarred."
-  cd "${WORKING_DIR}"
+  cd "${WORKING_DIR}" || exit
```
Comment (neomatrix369) [inline sbin/jtreg_prep.sh]: "${VAR}" same as ${VAR} or "$VAR"

File: sbin/jtreg_prep.sh [hunk 23]
```diff
@@ -95,7 +95,7 @@ fi
 if [ ! -d "${JAVA_DESTINATION}" ]; then
   mkdir -p "${JAVA_DESTINATION}"
 fi
-cd "${JAVA_DESTINATION}"
+cd "${JAVA_DESTINATION}"  || exit
```
Comment (neomatrix369) [inline sbin/jtreg_prep.sh]: Convention has changed on this line and below, "${VAR}" same as ${VAR} or "$VAR"

File: sbin/jtreg_prep.sh [hunk 24]
```diff
@@ -69,24 +69,24 @@ fi
...
   git reset --hard origin/$BRANCH
-  cd $WORKING_DIR
+  cd "$WORKING_DIR" || exit
 elif [ ! -d "${WORKING_DIR}"/$OPENJDK_REPO_NAME/.git ] ; then
   # If it doesn't exist, clone it
   echo "Didn't find any existing openjdk repository at WORKING_DIR (set to ${WORKING_DIR}) so cloning the source to openjdk"
   if [[ "${USE_SSH}" == true ]] ; then
     echo "git clone -b ${BRANCH} git@github.com:${REPOSITORY}.git $WORKING_DIR/$OPENJDK_REPO_NAME"
-    git clone -b ${BRANCH} git@github.com:${REPOSITORY}.git $WORKING_DIR/$OPENJDK_REPO_NAME
+    git clone -b ${BRANCH} git@github.com:"${REPOSITORY}".git "$WORKING_DIR/$OPENJDK_REPO_NAME"
```
Comment (neomatrix369) [inline sbin/jtreg_prep.sh]: This line the next few changed lines do not need "{$VAR}", just ${VAR} or "$VAR" should be enough

File: makejdk.sh (hunk 6 above)
Comment (tellison) [inline makejdk.sh]: But `${var}` is *not* the same as `"$var"`

See the [ref guide](https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html)

The parentheses are used to delimit the `var`  and the double quotes prevent globbing/word splitting of the var value.

So in this case we case we can use the quotes only, but if we don't put them in then the command will fail for a dir with spaces, e.g. try this
```
$ mkdir "test dir"
$ foo="test dir"
$ cd ${foo}
bash: cd: test: No such file or directory
$ cd "$foo"
[test dir]$ 
```

File: sbin/build.sh [hunk 25]
```diff
@@ -287,7 +292,7 @@ createOpenJDKTarArchive()
 
 stepIntoTargetDirectoryAndShowCompletionMessage()
 {
-  cd $TARGET_DIR
+  cd $TARGET_DIR  || return
```
Comment (neomatrix369) [inline sbin/build.sh]: No quotes around $TARGET?

File: sbin/jtreg.sh [hunk 26]
```diff
@@ -105,32 +106,32 @@ packageTestResultsWithJCovReports()
...
   pwd
 
-  cd $WORKING_DIR/$OPENJDK_REPO_NAME/build/$BUILD_FULL_NAME/
+  cd "$WORKING_DIR/$OPENJDK_REPO_NAME/build/$BUILD_FULL_NAME/" || exit
  
-  artifact=${JOB_NAME}-testoutput-with-jcov-reports
+  artifact="${JOB_NAME}-testoutput-with-jcov-reports"
   echo "Tarring and zipping the 'testoutput' folder into artefact: $artifact.tar.gz" 
-  tar -cvzf $WORKING_DIR/$artifact.tar.gz   testoutput/
+  tar -cvzf "$WORKING_DIR/$artifact.tar.gz"   testoutput/
```
Comment (neomatrix369) [inline sbin/jtreg.sh]: `v` will enlist all the files being tarred, do we want to show this to the end-user or record it in the logs - in that case its fine

File: makejdk.sh (hunk 4 above)
Comment (tellison) [inline makejdk.sh]: Changed, thanks

File: sbin/jtreg.sh [hunk 27]
```diff
@@ -105,32 +106,32 @@ packageTestResultsWithJCovReports()
...
   pwd
 
-  cd $WORKING_DIR/$OPENJDK_REPO_NAME/build/$BUILD_FULL_NAME/
+  cd "$WORKING_DIR/$OPENJDK_REPO_NAME/build/$BUILD_FULL_NAME/" || exit
  
-  artifact=${JOB_NAME}-jcov-results-only
+  artifact="${JOB_NAME}-jcov-results-only"
   echo "Tarring and zipping the 'testoutput/../jcov' folder into artefact: $artifact.tar.gz" 
-  tar -cvzf $WORKING_DIR/$artifact.tar.gz   testoutput/*/JTreport/jcov/
+  tar -cvzf "$WORKING_DIR/$artifact.tar.gz"   testoutput/*/JTreport/jcov/
```
Comment (neomatrix369) [inline sbin/jtreg.sh]: same `v` option here (verbose)

File: sbin/signalhandler.sh (hunk 8 above)
Comment (tellison) [inline sbin/signalhandler.sh]: Yep, I look forward to the contribution ;-)  I took the lazy way out of fixing the undefined vars, but reuse would be better.

File: sbin/signalhandler.sh (hunk 9 above)
Comment (tellison) [inline sbin/signalhandler.sh]: No, I really meant to use double quotes here to avoid whitespace issues

File: makejdk.sh (hunk 10 above)
Comment (tellison) [inline makejdk.sh]: Not sure it matters for a definition, but I've put them in.

**Comment by neomatrix369** [comment]
Overall a good PR especially it will help us decide on conventions and once merged, will make PR #53 look lighter, but I noticed the linter didn't help with suggesting a convention (or has it?). I have pointed out the areas where even though it make the strings safe, was applied differently than another area in the script.

I think we will have to come up and agree on guidelines - it does not matter so much what I pointed out, if they are acceptable or not. My (proposed) guidelines were:
- use `"$VAR"` in if conditions
- use `${VAR}` if it was part of another string/variable i.e. "The source is ${VAR}" or `cd ${VAR1}/{VAR2}`
- use `$VAR` in all other circumstances

We can discuss this / correct it and apply them to our upcoming PRs. :100:

File: makejdk.sh (hunk 15 above)
Comment (tellison) [inline makejdk.sh]: as above, really do need quotes.  If the house-style is {}'s too, that's fine I can add them.

File: makejdk.sh (hunk 17 above)
Comment (tellison) [inline makejdk.sh]: Should `$JTREG_TEST_SUBSETS` be splitting on spaces?  Not sure whether this is a single arg or set of args (in which case it should be an array).

File: sbin/build.sh (hunk 20 above)
Comment (tellison) [inline sbin/build.sh]: Now you are being silly ;-)  lol
I'll add some parentheses too .

File: sbin/build.sh (hunk 21 above)
Comment (tellison) [inline sbin/build.sh]: I think `$PARAMS`  is a single value, correct?  If not we should be using arrays.

File: sbin/jtreg_prep.sh (hunk 24 above)
Comment (tellison) [inline sbin/jtreg_prep.sh]: As above, the dirs definitely need quoting to get through to the command.
Will add {} for house style.

File: sbin/jtreg.sh (hunk 26 above)
Comment (tellison) [inline sbin/jtreg.sh]: I didn't change the `v` option?

**Comment by tellison** [comment]
I agree that adopting a convention would be good, and admit that I may have been inconsistent myself as I tried to fix things up.  The main goal here was to fix lint errors/warnings.  A separate pass to apply a consistent style throughout would be a good thing.  Perhaps a discussion in a new issue would be the right thing here?

**Comment by neomatrix369** [comment]
I think with this a great set of guidelines will come up - guided by the static analysis reports + our own decisions and due diligence

**Comment by neomatrix369** [comment]
@tellison agreed lets get this PR merged and raise/re-raise a new PR, in fact PR #53 covers a good amount of it, i can merge master once your changes are merged and we will have a PR with consistency in it (hopefully) - we can discuss the guidelines via that PR.

File: sbin/jtreg.sh (hunk 26 above)
Comment (neomatrix369) [inline sbin/jtreg.sh]: No you didnt just came to my notice
Can be a different PR

File: sbin/jtreg_prep.sh (hunk 22 above)
Comment (neomatrix369) [inline sbin/jtreg_prep.sh]: Ok lesson learn, will refer to the guide you mentioned

File: sbin/signalhandler.sh (hunk 8 above)
Comment (neomatrix369) [inline sbin/signalhandler.sh]: Done via PR #54

File: makejdk.sh (hunk 6 above)
Comment (neomatrix369) [inline makejdk.sh]: Thanks for the ref guide, i'll add it to my PR.

**Comment by neomatrix369** [comment]
LGTM is anyone else does not have any nit-picking interests on this PR

File: makejdk.sh (hunk 6 above)
Comment (gibfahn) [inline makejdk.sh]: >We had a discussion on one of the PR that "${}" is the same as ${}, i noted it for our guideline (see #54)

I think the conclusion there was that `${VAR}` is the same as `$VAR`, quotes are a different thing.

So it could be `"${VAR}"` or `"$VAR"`, but I guess that's a question for #54

File: git-hg/git-hg.sh (hunk 7 above)
Comment (gibfahn) [inline git-hg/git-hg.sh]: Agreed that `-eu` is a nice default, it's better to explicitly say `|| true` if you want to ignore an error.

File: makejdk.sh (hunk 15 above)
Comment (gibfahn) [inline makejdk.sh]: I think the house style is being decided in #54 , probably easiest to see how that goes and then update this (or land this as is and update in #54)...

**Comment by gibfahn** [comment]
>LGTM is anyone else does not have any nit-picking interests on this PR

setting `-eu` at the top seems reasonable.

File: sbin/jtreg.sh (hunk 26 above)
Comment (karianna) [inline sbin/jtreg.sh]: I removed the v from another script if we want to be consistent <shrug>

File: makejdk.sh (hunk 15 above)
Comment (neomatrix369) [inline makejdk.sh]: I would say filter all discussions and then apply it on top of #54 and use that as our final guidelines.
